### PR TITLE
Hunter 0.9.9.0

### PR DESCRIPTION
**Pull request by madmongo1**
merges upstream latest which fixes the non-compile error for gcc7.3
No idea why there are so many commits in here. As far as I am concerned, there are only 2 :/

### CODE REVIEW

**Comment by ruslo** [comment]
> No idea why there are so many commits in here. As far as I am concerned, there are only 2 :/

Branch `hunter-0.9.9-a2` based on latest release:
* https://github.com/g-truc/glm/releases/tag/0.9.9-a2

I guess your changes was not using it as a base, please do fix it.

**Comment by madmongo1** [comment]
The bug is fixed in a very recent commit in glm. I was just taking current
master, which works fine in my application.

I am lost as to how to get a commit into hunter_packages/glm

To fix it gdm for gcc7.3 all that is needed is to merge latest gdm master
into hunter and make a release tag, then update hunter to use latest gdm.

I'd love to help, but I don't know how to get a commit into the
hunter_packages repo.

basically it just needs a git cherry-pick

**Comment by ruslo** [comment]
> The bug is fixed in a very recent commit in glm

What is the exact commit? I can create branch basing on it.

**Comment by madmongo1** [comment]
https://github.com/g-truc/glm/commit/1c85265f97d82b11dc78d651156cd1815e01441c

**Comment by ruslo** [comment]
Branch created: https://github.com/hunter-packages/glm/commits/hunter-0.9.9-a2-1c85265

**Comment by madmongo1** [comment]
to be honest, there is already a later branch in hunter_packages/glm.

https://github.com/hunter-packages/glm/commits/hunter-0.9.9-a2

It might be easier to just issue a release tag of that one and update the
default config in hunter.

**Comment by ruslo** [comment]
> It might be easier to just issue a release tag of that one and update the default config in hunter.

There is no hunterization patches there.

**Comment by madmongo1** [comment]
glm does not have any dependencies. No need for hunterisation.

**Comment by ruslo** [comment]
> glm does not have any dependencies. No need for hunterisation.

Where was some patches applied, send them upstream then.

**Comment by madmongo1** [comment]
I don't think we are understanding each other.

upstream, the latest master is good code. It compiles with gcc7.3

hunter has no version of glm that will compile with gcc7.3 because the
latest hunter release targets an old version of glm.

So all that is needed is to pull the latest upstream/master into
hunter-packages/glm/hunter, create a release tag and then reference this
release tag in hunter's glm module code and the defaults/config.cmake

Then release another hunter. Then hunter_add_package(glm) will work with
gcc7.3. At the moment it does not.

Things I can do:
1. issue a pull request

Things I can't do:
1. make a new hunter-packages/glm release
2. make a patch for hunter (until I have a glm release and a sha1 for the
new glm release)
3. make a hunter release

The problem here for me is that this is a 2-step process, and I don't have
the permission to complete the first step.

hunter-package/glm is 7 months out of date.

**Comment by ruslo** [comment]
> upstream, the latest master is good code. It compiles with gcc7.3

There was a fix introduced, that's why fork exist. I think something about `*.pc` file patching:
* https://github.com/hunter-packages/glm/commit/f93e33a14faab3e210cb7103aa569fdb123285be

Please check that there is no other patches added.

> Things I can do: 1. issue a pull request

That's what I was asking. I've created new branch, please take a look at what patches was applied for previos version and cherry-pick them to newly created branch.

> Things I can't do: 1. make a new hunter-packages/glm release

I will do a release after successful merge.

> 2. make a patch for hunter (until I have a glm release and a sha1 for the new glm release)

I will do a release and you can create Hunter pull request with this release.

> 3. make a hunter release

I will do it.

> The problem here for me is that this is a 2-step process, and I don't have the permission to complete the first step.

There is no need to have any permissions, you only need to create pull request.

> hunter-package/glm is 7 months out of date.

And? I'm not a maintainer of all the packages. It just means that nobody was interested in update. You hit an issue and interested in update, congrats! :)

**Comment by madmongo1** [comment]
I have navigated to
https://github.com/hunter-packages/glm/tree/hunter-0.9.9-a2

I have pressed "pull request"

I get this:

[image: Inline images 1]

What now?

**Comment by ruslo** [comment]
> [image: Inline images 1]

I don't see anything

> What now?

https://help.github.com/categories/collaborating-with-issues-and-pull-requests/
